src/compiler-support.h

```diff
@@ -27,7 +27,7 @@
 
 // If control flow reaches the point of the WASM_UNREACHABLE(), the program is
 // undefined.
-#if __has_builtin(__builtin_unreachable)
+#if __has_builtin(__builtin_unreachable) && defined(NDEBUG)
```

Comment:
It means that it uses `__builtin_unreachable` only in release mode, and aborts in debug mode. Makes catching bugs easier.